.claude/commands/schema-migration.md

```diff
@@ -0,0 +1,447 @@
+# CockroachDB Schema Migration Construction
+
+You are helping to construct a schema migration in CockroachDB. Follow this comprehensive guide to ensure all components are properly implemented and tested.
+
+## Overview
+
+A CockroachDB schema migration involves multiple coordinated changes across several files to ensure consistency between the migration logic, bootstrap schemas, and version tracking. Use this when you need to modify system tables, add new features requiring schema changes, or update table storage parameters.
+
+## Version Numbering System
+
+CockroachDB uses a structured version numbering system:
+- **Major.Minor.Patch-Internal** format (e.g., 25.3.0-6)
+- **Internal versions must be even numbers** (2, 4, 6, 8...)
+- Each release starts with a `V{Major}_{Minor}_Start` version
+- Development versions use `{Major}.{Minor-1}.Internal` format during development
+
+**Example for v25.4 development:**
+- `V25_4_Start: {Major: 25, Minor: 3, Internal: 2}`
+- `V25_4_YourFeature: {Major: 25, Minor: 3, Internal: 4}` (next even number)
+- `V25_4_AnotherFeature: {Major: 25, Minor: 3, Internal: 6}` (next even number)
+
+## Step-by-Step Process
+
+### 1. Add a New Cluster Version
+
+**File:** `pkg/clusterversion/cockroach_versions.go`
+
+1. Add a new version constant in the version enum **before** the `// Step (1)` comment:
+```go
+// V25_4_YourFeatureName describes what your migration does
+// and why it's needed for clarity.
+V25_4_YourFeatureName
+```
+
+2. Add the corresponding version mapping **before** the `// Step (2)` comment:
+```go
+V25_4_YourFeatureName: {Major: 25, Minor: 3, Internal: 8}, // Use next even number
+```
+
+**Important:** Always use the next available even Internal number. Check existing versions to see what's already used.
+
+### 2. Create the Migration File
```

Comment:
I personally prefer to reverse the order here: I only add the upgrade after I've modified the bootstrap catalog and tested my feature against the updated schema, since I often need to iterate on the schema a little as I do. Once I've arrived at a schema that works for my feature, I add the upgrade for existing clusters once, rather than doing that first and needing to keep updating it as I tweak my new schema.